### PR TITLE
Add error constants

### PR DESCRIPTION
**Pull request by kasperisager**
The current errors could likely benefit from a little cleanup as some of them are quite similar.

### CODE REVIEW

**Comment by kasperisager** [comment]
@mafintosh This is ready for review.

File: lib/errors.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  static HOLEPUNCH_INVALID (msg = 'Invalid holepunch payload') {
+    return new DHTError(msg, 'HOLEPUNCH_INVALID', DHTError.HOLEPUNCH_INVALID)
+  }
+
+  static HOLEPUNCH_PROBE_TIMEOUT (msg = 'Holepunching probe did not finish in time') {
+    return new DHTError(msg, 'HOLEPUNCH_PROBE_TIMEOUT', DHTError.HOLEPUNCH_PROBE_TIMEOUT)
+  }
+
+  static HOLEPUNCH_RANDOMIZED_NATS (msg = 'Both remote and local NATs are randomized') {
+    return new DHTError(msg, 'HOLEPUNCH_RANDOMIZED_NATS', DHTError.HOLEPUNCH_RANDOMIZED_NATS)
```
Comment (mafintosh) [inline lib/errors.js:36]: can we call this one HOLEPUNCH_DOUBLE_RANDOMIZED_NATS to be more clear

**Comment by mafintosh** [comment]
@kasperisager looks great other than my nit

**Comment by mafintosh** [comment]
Awesome, let me add @andrewosh for an English speaker speaker review

**Review by andrewosh — APPROVED** [review]
English language lookin' good

**Pull request merged**